### ISSUE TITLE
Cookies are not set during node-fetch redirect

### ISSUE THREAD

**Issue by mvindahl** [valeriangalliat/fetch-cookie#6]
This is due to redirects happening internally in node-fetch, and not returning to the wrapping fetch-cookie until redirection has ended. There is some discussion on this in the node-fetch issue tracker (See: https://github.com/bitinn/node-fetch/issues/94). It seems like their solution has been to add the "redirect: 'manual'" option, allowing wrapping libraries such as node-fetch to handle redirects on their own.

This also points to how it could be made to work.

I have two ideas:
1) I could extend fetch-cookie to take advantage of this option. It's certainly doable but it would also make the code less succinct and it would couple the library tightly to node-fetch instead of being a more generic fetch wrapper. Thus, it may pollute the design and goals of fetch-cookie. And, no matter how careful I am, there is always the risk of breaking stuff.
2) I could create a separate project node-fetch-cookie as an additional wrapper. It would require fetch-cookie, and it would take a node-fetch instance in its constructor in the same way that fetch-cookie does. As the name might hint, it could be a drop-in replacement for fetch-cookie, but specifically tailored to wrap node-fetch, and without the cookie-and-redirect issue.

I kind of like the second one better as it seems to be more in line with the "do one thing well" philosophy.

I would, however, appreciate any thoughts on the matter.

**Comment by valeriangalliat** [comment]
Hi @mvindahl, thanks for taking the time to report this issue and propose solutions, I'm sorry for taking that long to answer.

I tend to prefer the second option because I like that this library is not tied to any `fetch` implementation, but since the browser implementation already handle cookies if I recall correctly, `fetch-cookie` is more likely going to be used with `node-fetch` anyway... but on the other hand, I don't want to disallow the use of other Node `fetch` implementations.

So I prefer your second idea, which makes it clear that it's coupled to `node-fetch` for a feature that, if I understand properly, cannot be implemented with only the generic `fetch` specification.

If you'd like, I'm thinking it would make sense to have it in this repo and have something like:

```js
var fetch = require('fetch-cookie/node-fetch')(require('node-fetch'))
```

What do you think? If you want to add it in this repo, I'll be glad to give you commit access. Otherwise, I definitely encourage you to create `node-fetch-cookie`. :)

**Comment by mvindahl** [comment]
Hi Valérian,

NP with the response time.

I also liked the second option better. Especially after having detailed them both. So actually, I already have a working implementation in my codebase which can easily be extracted.

I like your idea of including the functionality into your project. I think this is better than creating yet another featherweight npm module.

I think the best approach would be for me to fork your project and create a pull request for you to review and approve. This will ensure that you accept the quality of the solution since you will be the official owner of the code going forward. I'll have to allocate a slot of my spare time so I'm not exactly sure when it'll be but I hope to do it soon.

**Comment by valeriangalliat** [comment]
That sounds good, thank you!

**Comment by valeriangalliat** [comment]
Fixed with #9.
